apps/backend/src/controllers/neighborhoods.ts

```diff
@@ -3,95 +3,102 @@ import { Neighborhood } from '@prisma/client';
+  const loggedUserID = req.loggedUserId as number;
+
+  const isUserAdminOfNeighborhood = await neighborhoodServices
+    .isUserAdminOfNeighborhood(loggedUserID, neighborhoodID);
+
+  if (!isUserAdminOfNeighborhood) {
+    return res.status(403).send({ error: 'User is not the admin of this neighborhood' });
   }
+
+  const deletedNeighborhood = await neighborhoodServices.deleteNeighborhood(neighborhoodID);
+  return res.status(200).send(`Neighborhood '${deletedNeighborhood.name}' has been deleted.`);
 }));
 
-neighborhoodsRouter.put('/:id', middleware.userExtractor, catchError(async (req: CustomRequest, res: Response) => {
-  if (!(await routeHelpers.isLoggedInAdmin(req))) {
-    return res.status(403).send({ error: 'User does not have edit rights for this neighborhood.' });
+// Since update routes are not critical, not spending too much time on it
+neighborhoodsRouter.put('/:id', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const neighborhoodID = Number(req.params.id);
+
+  // LoginValidator ensures that loggedUserId is present
+  const loggedUserID = req.loggedUserId as number;
+
+  const isUserAdminOfNeighborhood = await neighborhoodServices
+    .isUserAdminOfNeighborhood(loggedUserID, neighborhoodID);
+
+  if (!isUserAdminOfNeighborhood) {
+    return res.status(403).send({ error: 'User is not the admin of this neighborhood' });
   }
 
   const data = req.body;
   const updatedNeighborhood: Neighborhood = await prismaClient.neighborhood.update({
     where: { id: +req.params.id },
     data,
   });
-
   return res.status(200).send(`Neighborhood '${updatedNeighborhood.name}' has been updated.`);
 }));
 
-neighborhoodsRouter.post('/', middleware.userExtractor, catchError(async (req: CustomRequest, res: Response) => {
-  const userId: number = req.user?.id as number; // user.id shoul be extracted from the middleware
-  req.body.admin_id = userId; // adding user_id as admin_id to request.body
-  const createNeighborhoodData = await routeHelpers.generateCreateNeighborhoodData(req.body);
+neighborhoodsRouter.post('/', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const loggedUserID = req.loggedUserId as number; // loggedUserId extracted by middleware
+  req.body.admin_id = loggedUserID; // adding user_id as admin_id to request.body
+
+  const createNeighborhoodData: CreateNeighborhoodData = await neighborhoodServices
+    .parseCreateNeighborhoodData(req.body);
 
-  const newNeighborhood: Neighborhood = await prismaClient.neighborhood
-    .create({ data: createNeighborhoodData });
+  const newNeighborhood: Neighborhood = await neighborhoodServices
+    .createNeighborhood(createNeighborhoodData);
 
-  await routeHelpers.connectUsertoNeighborhood(userId, newNeighborhood.id);
+  await neighborhoodServices.connectUserToNeighborhood(loggedUserID, newNeighborhood.id);
```

Comment:
Since whenever a neighbourhood is created the current user will be connected to it, should we move this functionality into the `createNeighborhood` service?

Reply:
Yeah, but then I thought that would be 1 function doing too many things. So I let just `createNeighborhood` service to create a neighborhood and used another service to make user join neighborhood.

Reply:
Agree with hard-coding the type for the above.